### DISCUSSION TITLE
Is there a way to start a web-server or an electron (JS) application and view it?

### DISCUSSION THREAD

**Discussion by HalfTurk** [coder/code-server#4224] [Q&A]
Does anyone know how I could run a web app and be able to view it within code-server or externally on Safari?

**Comment by jsjoeio** [comment]
Yes! We have some docs around this [here](https://coder.com/docs/code-server/v3.12.0/guide#accessing-web-services). Let me know if that helps!

**Reply by HalfTurk** [reply]
Thanks so much! But now, is it possible for me to gain administrative permissions on Heroku inside the actual terminal? Like instead of me having to restart it and edit the Dockerfile each and every time, how could I get administrative permissions just by signing on and then doing my usual? I already tried giving myself sudo by editing and adding the command but it doesn’t seem to work. Any ideas? (Pardon my English. I’m tired and I need some rest!)

**Reply by jsjoeio** [reply]
Hmm...I remember seeing a thread saying that `sudo` or admin perms weren't possible on Heroku 🤔 

What are you trying to do? @bpmct or myself might be able to help

**Reply by HalfTurk** [reply]
I can’t seem to download lots of packages without sudo. I know I’d have to use the Dockerfile to install them but it doesn’t let me use some commands without sudo, if you know what I mean? I can’t really explain it to you if you don’t get it but just see that I can’t execute some commands without having sudo privileges. If it doesn’t work, I guess I’ll have to figure out how to automatically give the user sudo when they login. It doesn’t make a lot of sense why I can’t find a way to make it work but I’m sure there will be some way to make it work :). Thanks for the help! I appreciate it :)

**Reply by jsjoeio** [reply]
Ahhh! I see what you mean. Your Heroku environment is basically a Docker container (which has code-server in the image) and the only way to install packages is by adding them to your Dockerfile.

I'm guessing this is because it spins up and down depending on when you're using it.

I was actually having this conversation with @bpmct the other day about what to put in my Docker image vs. my install script.

Probably not new info but on S/O, seems like we can't get around this because you can't run as root on Heroku. See [here](https://stackoverflow.com/a/45421218/3015595) :(

**Reply by HalfTurk** [reply]
You know, I reckon it could be possible somehow. If I can do it, although I’m underage, I want a internship one day. Lol. Thanks for the help! I’ll reopen this if I can do it. But if I can’t do it, then you right. There’s got to be a way to bypass it. I can definitely see why they don’t give sudo permissions to *normal* people for reasons which are self explanatory. (Btw, code-server saved my life. Without coding, especially with a 5 pound laptop on me at most times, it helps for me to code on the go.) Aside from that, it’s amazing that you and your team of devs or all your colleagues made this. I hope development is going smooth and that the project continues throughout the years. Can’t thank you enough. This quickly evolved from a question into a thank you letter 😂. You can close the ticket or if you have something to add or another idea, I’d love to hear it :)

**Reply by jsjoeio** [reply]
> You know, I reckon it could be possible somehow

Well if you figure it out, share with the community!

> If I can do it, although I’m underage, I want a internship one day

I think you'll _easily_ get one by the time you're of age or ready. Think of how many years of experience you'll have 😂

> (Btw, code-server saved my life. Without coding, especially with a 5 pound laptop on me at most times, it helps for me to code on the go.) Aside from that, it’s amazing that you and your team of devs or all your colleagues made this. I hope development is going smooth and that the project continues throughout the years. Can’t thank you enough. This quickly evolved from a question into a thank you letter 😂.

That's wonderful to hear! Thank you for saying all that. The last part made me laugh, I shared with the rest of the team 🤣

Thanks for being part of the community! And if you ever want to share with us something cool you build, drop a post in the Show and Tell section and we'll see it :) 

See you around GitHub!

**Reply by GirlBossRush** [reply]
@KappasDEV, thank you so much for your kind words. There’s more in store for code-server soon!
